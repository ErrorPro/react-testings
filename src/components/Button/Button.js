import React, { Component } from 'react';
import PropTypes from 'prop-types';

class Button extends Component {
  state = {
    counter: 0,
  }

  render() {
    const { label, onClick } = this.props;
    const { counter } = this.state;

    return (
      <button
        onClick={() => {
          this.setState({ counter: counter + 1 }, () => {
            onClick(this.state.counter);
          });
        }}
      >
        {label} {counter}
      </button>
    )
  }
};

Button.propTypes = {
  onClick: PropTypes.func.isRequired,
  label: PropTypes.string,
};

export default Button;
