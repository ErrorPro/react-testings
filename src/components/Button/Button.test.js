import React from 'react';
import Button from './Button';
import renderer from 'react-test-renderer';
import Enzyme, { mount } from 'enzyme';
import Adapter from 'enzyme-adapter-react-16';

Enzyme.configure({ adapter: new Adapter() });

test('renders button with passed props', () => {
  const component = renderer.create(
    <Button onClick={() => {}} label="this is test label" />
  );

  expect(component.toJSON()).toMatchSnapshot();
});

test('renders with 0 as an initial state', () => {
  const component = renderer.create(
    <Button onClick={() => {}} label="this is test label" />
  );
  const instance = component.getInstance();

  expect(instance.state.counter).toBe(0);
});

test('onClick function is being called once with state counter', () => {
  const fn = jest.fn();
  const component = renderer.create(
    <Button onClick={fn} label="this is test label" />
  );
  const instance = component.getInstance();

  instance.props.onClick();

  expect(fn.mock.calls.length).toBe(1);
});

test('onClick function is called with passed counter value', () => {
  const fn = jest.fn();
  const component = mount(
    <Button onClick={fn} label="this is test label" />
  );

  component.find('button').simulate('click');

  expect(component.state('counter')).toBe(1);
  expect(fn.mock.calls[0][0]).toBe(1);
});
