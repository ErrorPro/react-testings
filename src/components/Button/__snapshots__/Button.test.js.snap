// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`renders button with passed props 1`] = `
<button
  onClick={[Function]}
>
  this is test label
   
  0
</button>
`;
